DataSetExplorer/Startup.cs

```diff
@@ -14,11 +17,32 @@ namespace DataSetExplorer
 {
     public class Startup
     {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddSingleton(Configuration);
+
+            services.AddAutoMapper(typeof(Startup));
+
             services.AddControllers();
+
             services.AddDbContext<DataSetExplorerContext>(opt =>
                 opt.UseNpgsql(CreateConnectionStringFromEnvironment()));
+
+            services.AddScoped<IDataSetCreationService, DataSetCreationService>();
+            services.AddScoped<ICodeRepository, GitCodeRepository>();
+            services.AddSingleton<IDataSetRepository, DataSetDatabaseRepository>();
```

Comment:
Should be AddScoped instead of AddSingleton

Reply:
I changed to _Transient_ now, because cannot update dataset after cloning repo with _Scoped_.

Reply:
This issue shouldn't be happening. @simsimkic can you pull the latest changes from this branch and see what is going on when you get a chance in the next few days?